### PR TITLE
feat: Added KQL for ExpressRoute FastPath Connections

### PR DESCRIPTION
**Pull request by DaFitRobsta**
# Overview/Summary

Added KQL for ExpressRoute FastPath Connections.
The query will find all ExpressRoute Connections, verify if the Virtual Network Gateway is of the right SKU level to have the option to enable FastPath. It returns both ExpressRoute Gateway Bypass and Private Endpoint/Link FastPath properties to determine if both are enabled. This query will also return ALI, SAP, AVS, etc. ExpressRoute Circuit Connections as well.

image

### Breaking Changes

None

## As part of this pull request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Review by oZakari — APPROVED** [review]
Nice work, @DaFitRobsta!

File: azure-resources/Network/connections/kql/f6a14b32-a727-4ace-b5fa-7b1c6bdff402.kql [hunk 1]
```diff
@@ -1,2 +1,20 @@
...
+    resources
+    | where type =~ "Microsoft.Network/virtualNetworkGateways"
+    | where properties.sku.name in~ ("ErGw3AZ", "UltraPerformance")
+    | extend gatewayId = tostring(id)
+) on gatewayId
+| extend erGatewayBypass = tobool(properties.expressRouteGatewayBypass)
+| extend privateLinkFastPath = tobool(properties.enablePrivateLinkFastPath)
+| where not(erGatewayBypass) or not(privateLinkFastPath)
+| project recommendationId = "f6a14b32-a727-4ace-b5fa-7b1c6bdff402", id, name, tags,
+    param1 = iff(erGatewayBypass, "✅ Enabled: Gateway Bypass", "❌ Disabled: Gateway Bypass"),
```
Comment (ejhenry) [inline azure-resources/Network/connections/kql/f6a14b32-a727-4ace-b5fa-7b1c6bdff402.kql:18]:
```suggestion
    param1 = iff(erGatewayBypass, "Enabled: Gateway Bypass", "Disabled: Gateway Bypass"),
```

File: azure-resources/Network/connections/kql/f6a14b32-a727-4ace-b5fa-7b1c6bdff402.kql [hunk 2]
```diff
@@ -1,2 +1,20 @@
...
+    param2 = iff(privateLinkFastPath, "✅ Enabled: PE FastPath", "❌ Disabled: PE FastPath"),
```
Comment (ejhenry) [inline azure-resources/Network/connections/kql/f6a14b32-a727-4ace-b5fa-7b1c6bdff402.kql:19]:
```suggestion
    param2 = iff(privateLinkFastPath, "Enabled: PE FastPath", "Disabled: PE FastPath"),
```

File: azure-resources/Network/connections/kql/f6a14b32-a727-4ace-b5fa-7b1c6bdff402.kql [hunk 3]
```diff
@@ -1,2 +1,20 @@
...
+    | where properties.sku.name in~ ("ErGw3AZ", "UltraPerformance")
+    | extend gatewayId = tostring(id)
+) on gatewayId
+| extend erGatewayBypass = tobool(properties.expressRouteGatewayBypass)
+| extend privateLinkFastPath = tobool(properties.enablePrivateLinkFastPath)
+| where not(erGatewayBypass) or not(privateLinkFastPath)
+| project recommendationId = "f6a14b32-a727-4ace-b5fa-7b1c6bdff402", id, name, tags,
+    param1 = iff(erGatewayBypass, "Enabled: Gateway Bypass", "Disabled: Gateway Bypass"),
+    param2 = iff(privateLinkFastPath, "Enabled: PE FastPath", "Disabled: PE FastPath"),
+    param3 = "More info: https://learn.microsoft.com/en-us/azure/expressroute/about-fastpath"
```
Comment (ejhenry) [inline azure-resources/Network/connections/kql/f6a14b32-a727-4ace-b5fa-7b1c6bdff402.kql:20]:
```suggestion
```

**Review by ehaslett — APPROVED** [review]
LGTM - nice work!!

**Review by oZakari — APPROVED** [review]

**Pull request merged**